frontend/src/metabase/writeback/types/formSettingsTypes.ts

```diff
@@ -0,0 +1,45 @@
+export type FormType = "inline" | "modal";
+export type FieldType = "text" | "number" | "date" | "category";
+export type InputType =
+  | "text"
+  | "longtext"
+  | "number"
+  | "date"
+  | "datetime"
+  | "monthyear"
+  | "quarteryear"
+  | "dropdown"
+  | "inline-select";
```

Comment:
I don't think this is necessarily the final list, but I think it's pretty close.